peps/pep-9999.rst

```diff
@@ -0,0 +1,632 @@
+of the problems that we are trying to solve. It was decided that
+we need to agree on the current problems with the C API, before
+we are able to evaluate any of the proposed solutions. We
+therefore created the
+[`capi-workgroup <https://github.com/capi-workgroup/problems/issues/>`__]
+repository on GitHub in order to collect everyone's ideas on that
+question.
+
+Over 60 different issues were created on that repository, each
+describing a problem with the C API. They were categorized and
+a number of recurring themes were identified. The sections below
+mostly correspond to these themes, and each contains a combined
+description of the issues raised in that category, along with
+links to the individual issues. In addition, we included a section
+that aims to identify the different stakeholders of the C API,
+and the particular requirements that each of them has.
+
+
+C API Stakeholders
+==================
+
+As mentioned in the introduction, the C API was originally
+created as the internal interface between CPython's
+interpreter and the Python layer. It was later exposed as
+a way for third party developers to extend and embed Python
+programs. Over the years, new types of stakeholders emerged,
+with different requirements and areas of focus. This section
+describes this complex state of affairs in terms of the
+actions that different stakeholders need to perform through
+the C API.
+
+Universal Actions for All Stakeholders
+--------------------------------------
+
+There are actions which are generic, and required by
+all types of API users:
+
+* Define functions and call them
+* Create classes and instantiate them
+* Create instances of builtin and user defined types
+  and perform operations on them
+* Introspect objects, including types, instances, and functions.
+* Raise and handle exceptions
+* Import modules
+* Manage threads
+* Manage sub-interpreters
+* Handle and send signals
+
+The following sections look at the unique requirements of various stakeholders.
+
+Extension writers
+-----------------
+
+Extension writers are the traditional users of the C API. Their requirements
+are the universal actions listed above.
+
+Authors of Embedded Python Applications
+---------------------------------------
+
+Applications with an embedded Python interpreter. Examples are
+`Blender <https://docs.blender.org/api/current/info_overview.html>`__ and
+`OBS <https://obsproject.com/wiki/Getting-Started-With-OBS-Scripting>`__.
+
+They need to be able to:
+
+* Configure the interpreter (import paths, inittab, sys.argv, ...)
+* Interact with the execution model and program lifetime, including
+  clean interpreter shutdown and restart
+* Represent complex data models in a way Python can use without
+  having to create deep copies.
+* Provide and import frozen modules.
+* Run multiple independent interpreters (in particular, when embedded
+  in a library that wants to avoid global effects).
```

Comment:
```suggestion
* Configure the interpreter (import paths, inittab, ``sys.argv``, etc.).
* Interact with the execution model and program lifetime, including
  clean interpreter shutdown and restart.
* Represent complex data models in a way Python can use without
  having to create deep copies.
* Provide and import frozen modules.
* Run multiple independent interpreters (in particular, when embedded
  in a library that wants to avoid global effects).
```